materialize/extensions.go

```diff
@@ -153,6 +153,141 @@ func (m *ApplyResponse) UnmarshalJSON(b []byte) error {
 	return jsonpb.Unmarshal(bytes.NewReader(b), m)
 }
 
+func (m *TransactionRequest) Validate() error {
+	var count int
+	if m.Open != nil {
+		if err := m.Open.Validate(); err != nil {
+			return pb.ExtendContext(err, "Open")
+		}
+		count += 1
+	}
+	if m.Load != nil {
+		if err := m.Load.Validate(); err != nil {
+			return pb.ExtendContext(err, "Load")
+		}
+		count += 1
+	}
+	if m.Prepare != nil {
+		if err := m.Prepare.Validate(); err != nil {
+			return pb.ExtendContext(err, "Prepare")
+		}
+		count += 1
+	}
+	if m.Store != nil {
+		if err := m.Store.Validate(); err != nil {
+			return pb.ExtendContext(err, "Store")
+		}
+		count += 1
+	}
+	if m.Commit != nil {
+		if err := m.Commit.Validate(); err != nil {
+			return pb.ExtendContext(err, "Commit")
+		}
+		count += 1
+	}
+
+	if count != 1 {
+		return pb.NewValidationError("expected one of Open, Load, Prepare, Store, Commit")
+	}
```

Comment:
Is this a common go idiom for treating a struct like an enum?

Reply:
I'm not sure I've seen enough of it to call it idiomatic. Mostly I don't see it checked, and it makes me ☹️ . 

Generics are finally coming to Go, but I'd dearly love me some Rust-like algebraic enums and pattern matching....